config/src/base.rs

```diff
@@ -46,3 +60,135 @@ impl Default for BaseConfig {
         }
     }
 }
+
+impl BaseConfig {
+    /// Load local network configuration from Yaml file.
+    pub fn from_yaml_file() -> anyhow::Result<Self> {
+        let path = sui_config_dir()?.join(ETH_LOCAL_NETWORK_CONFIG);
+
+        let file_content = fs::read_to_string(path)?;
+        let mut config: BaseConfig = serde_yaml::from_str(&file_content)?;
+
+        if config.max_checkpoint_age == 0 {
+            config.max_checkpoint_age = CHECKPOINT_AGE_14_DAYS;
+        }
```

Comment:
is this a shorthand for deafult value? if so please explain in comment

Reply:
changed it to use a derive default function when deserializing using serde.
I've added the explanation there (config/src/utils.rs at the end).